couchdb/server/helpers.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+#
+from inspect import getsource
+from textwrap import dedent
+from types import FunctionType
+
+from couchdb import util
+
+
+def maybe_extract_source(fun):
+    if isinstance(fun, FunctionType):
+        return dedent(getsource(fun))
```

Comment:
When i was on the bus, re-think this patch again. Found the `.strip()` do not make sense, it should be `strip('\n')`. And i just run the test again, notice that the string src won't invoke `dedent`.